File: Leetcode_Medium/_167. Two Sum II - Input Array Is Sorted/Solution.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        # Initialize two pointer at the 
        # start and the end of the list
        left, right = 0 , len(numbers) - 1
        while (left < right):
            sum_number = numbers[left] + numbers[right]
            if (sum_number < target):
                left += 1
            elif(sum_number == target):
                return [left + 1, right + 1]
            else:
                right -= 1

File: Leetcode_Medium/_167. Two Sum II - Input Array Is Sorted/Solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        // Initialize two pointer at the start and end
        // of the array
        int left = 0;
        int right = numbers.length - 1;
        // The loop continue until left == right
        while (left < right){
            // Create a sum integer of left and right value
            int sumNum = numbers[left] + numbers[right];
            // If the value are smaller than the target
            // Also knowning that the array is sorted, we have 
            // to increase the index of left to increase the total
            // sumNum value
            if (sumNum < target){
                left ++;
            }else if(sumNum == target){
                int[] resArr = new int[2];
                resArr[0] = left+1;
                resArr[1] = right + 1;
                return resArr;
            } else{
                // If the value are bigger, then we reduce
                // the right pointer to make the total sum smaller
                right--;
            }
        }
        return new int[2];
        
    }
}